language.py

```diff
@@ -1196,25 +1220,31 @@ def to_lean_str(self):
             self.v3.to_lean_str()
 
 class LExprOp(LExpr):
-  def __init__(self, op, v):
+  def __init__(self, op, bw, v):
     assert isinstance(op, str)
+    assert isinstance(bw, str) or isinstance(bw, int)
     assert isinstance(v, LVar)
     self.op = op
+    self.bw = bw
     self.v = v
   def __repr__(self):
-    return "op:" + self.op + " " + self.v.to_lean_str()
+    if self.op.find("const") == -1:
+      return "op:" + self.op + " " + str(self.bw) + " " + self.v.to_lean_str()
+    else:
+      return "op:" + self.op + " " + self.v.to_lean_str()
 
   def to_lean_str(self):
     return str(self)
 
 class ToLeanState:
   def unit_index(self):
-    return LVar(9999)
+    return LVar(0) # 0 should never be assigned
 
   def __init__(self):
     self.assigns = []
     self.constant_names = []
     self.varmap = {}
+    self.bitwidths = {}
```

Comment:
@goens I am very uncomfortable with this, as it feels to me like it would make maintainance quite hard. We already have an AST. Can we compute store the bitwidths in the `LExpr` at the appropriate location?

Reply:
I am surprised that you're very uncomfortable with it, and I don't see why this makes maintenence significantly harder. I'm not saying it's not, I just don't understand why, so I would like to understand :) can you explain why you think this changes much? In the end, it's just using your own design with a state object too. This seems to fit perfectly in the whole `ToLeanState` buisiness. If you were doing everything on the `LExpr` world I would see the point more.